lib/Doctrine/DBAL/Driver/ResultStatement.php

```diff
@@ -28,66 +81,44 @@ public function columnCount();
     /**
      * Sets the fetch mode to use while iterating this statement.
      *
-     * @param integer $fetchMode The fetch mode must be one of the PDO::FETCH_* constants.
+     * @param integer $fetchMode The fetch mode must be one of the self::FETCH_* constants.
      * @param mixed   $arg2
      * @param mixed   $arg3
      *
      * @return boolean
      *
-     * @see PDO::FETCH_* constants.
+     * @see self::FETCH_* constants.
      */
     public function setFetchMode($fetchMode, $arg2 = null, $arg3 = null);
 
     /**
      * Returns the next row of a result set.
      *
      * @param int|null $fetchMode    Controls how the next row will be returned to the caller.
-     *                               The value must be one of the \PDO::FETCH_* constants,
-     *                               defaulting to \PDO::FETCH_BOTH.
-     * @param int $cursorOrientation For a ResultStatement object representing a scrollable cursor,
-     *                               this value determines which row will be returned to the caller.
-     *                               This value must be one of the \PDO::FETCH_ORI_* constants,
-     *                               defaulting to \PDO::FETCH_ORI_NEXT. To request a scrollable
-     *                               cursor for your ResultStatement object, you must set the \PDO::ATTR_CURSOR
-     *                               attribute to \PDO::CURSOR_SCROLL when you prepare the SQL statement with
-     *                               \PDO::prepare().
-     * @param int $cursorOffset      For a ResultStatement object representing a scrollable cursor for which the
-     *                               cursorOrientation parameter is set to \PDO::FETCH_ORI_ABS, this value
-     *                               specifies the absolute number of the row in the result set that shall be
-     *                               fetched.
-     *                               For a ResultStatement object representing a scrollable cursor for which the
-     *                               cursorOrientation parameter is set to \PDO::FETCH_ORI_REL, this value
-     *                               specifies the row to fetch relative to the cursor position before
-     *                               ResultStatement::fetch() was called.
+     *                               The value must be one of the self::FETCH_* constants,
+     *                               defaulting to self::FETCH_BOTH.
+     * @param array    $args         Optional mode-specific arguments.
      *
      * @return mixed The return value of this method on success depends on the fetch mode. In all cases, FALSE is
      *               returned on failure.
      *
-     * @see PDO::FETCH_* constants.
+     * @see self::FETCH_* constants.
      */
-    public function fetch($fetchMode = null, $cursorOrientation = \PDO::FETCH_ORI_NEXT, $cursorOffset = 0);
+    public function fetch($fetchMode = null, ...$args);
```

Comment:
This signature change is BC breaking in a way that is impossible to fix in a code that targets DBAL 2.x and 3.x. These types of BC breaks are really annoying for consumers and should be considered carefully. https://3v4l.org/nFacb